api/v1/openapi.json

```diff
@@ -0,0 +1,1115 @@
+                                        }
+				}
+			},
+			"delete": {
+				"description": "Delete an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{ "$ref": "#/parameters/endpoint-id" },
+					{ "$ref": "#/parameters/endpoint-id-type" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Endpoint"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			}
+		},
+                "/endpoint/list": {
+			"get": {
+				"summary": "Retrieve list of all endpoints",
+				"tags": [ "endpoint" ],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+							"type": "array",
+						        "items": {
+								"$ref": "#/definitions/Endpoint"
+							}
+                                                }
+					}
+				}
+			}
+		},
+                "/endpoint/config": {
+			"get": {
+				"summary": "Retrieve endpoint configuration",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{
+						"name": "id",
+						"description": "Endpoint ID",
+						"in": "query",
+						"required": true,
+						"type": "integer"
+					}
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/ConfigurationResponse"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+                        "put": {
+                                "summary": "Update mutable endpoint configuration",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{
+						"name": "id",
+						"description": "Endpoint ID",
+						"in": "query",
+						"required": true,
+						"type": "integer"
+					},
+					{
+						"name": "configuration",
+						"in": "body",
+						"required": true,
+						"schema": {
+							"$ref": "#/definitions/ConfigurationMap"
```

Comment:
Definitely something we should add, this is currently not possible. The API can be extended with this later on though without breaking compatibility.